### PR TITLE
ElasticMapper Tool (contrib)

### PR DESCRIPTION
**Pull request by SYNchroACK**
@aaronkaplan as we talked :+1: 
 
**Mapping Ouput:** [mapping.txt](https://github.com/certtools/intelmq/files/1608169/mapping.txt)

### CODE REVIEW

**Comment by robcza** [comment]
@SYNchroACK mapping seems nice, I'd also like to see Elasticsearch versions that would accept it as the mapping changes and will eventually be completely removed in future versions

File: contrib/elasticsearch/elasticmapper [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+import argparse
+import json
+import sys
+
+try:
+    from elasticsearch import Elasticsearch
+except:
+    print("Please install ElasticSearch 'pip install elasticsearch'")
+    sys.exit(0)
```
Comment (wagner-certat) [inline contrib/elasticsearch/elasticmapper]: Exit code should be non-zero.

File: contrib/elasticsearch/elasticmapper [hunk 2]
```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import sys
+
+try:
+    from elasticsearch import Elasticsearch
+except:
+    print("Please install ElasticSearch 'pip install elasticsearch'")
```
Comment (wagner-certat) [inline contrib/elasticsearch/elasticmapper]: Should be on stderr. (with `file=sys.stderr`)

**Comment by aaronkaplan** [comment]
I just noticed that the Travis CI checks failed
Will test this script nevertheless independently

**Comment by SYNchroACK** [comment]
@wagner-certat 

> I don't understand what the script does,

b0fb0c8b69680ce53fd6086407c29513f24a7820

> but the error handling could be better. What if there's a new Harmonization type which is not defined here? Could be either a default fallback along with a warning message (on stderr, again) or a human-understandable error.

a09d25f954aea7486b1c8a5cd8239f0130864285

> Does -h work?

Yes.

* * *

@aaronkaplan 

> I just noticed that the Travis CI checks failed
> Will test this script nevertheless independently

I think was related to some Travis issue, not IntelMQ. I guess they already solved it.

**Comment by wagner-certat** [comment]
> @aaronkaplan
>
>   > I just noticed that the Travis CI checks failed
>   > Will test this script nevertheless independently
>
> I think was related to some Travis issue, not IntelMQ. I guess they already solved it.

If you mean this:

```
======================================================================
FAIL: Tests if the parse_logline() function parses syslog correctly.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/certtools/intelmq/intelmq/tests/lib/test_utils.py", line 139, in test_parse_logline_syslog
    'message': 'Something went wrong'}, actual)
AssertionError: {'log[25 chars] '2017-02-22T10:17:10', 'message': 'Something [50 chars]tor'} != {'log[25 chars] '2018-02-22T10:17:10', 'message': 'Something [50 chars]tor'}
  {'bot_id': 'malware-domain-list-collector',
-  'date': '2017-02-22T10:17:10',
?              ^
+  'date': '2018-02-22T10:17:10',
?              ^
   'log_level': 'ERROR',
   'message': 'Something went wrong'}
```
That was a bug in the tests which I fixed in develop and maintenance yesterday.

**Pull request merged**